packages/client/test/integration/GroupKeyPersistence.test.ts

```diff
@@ -145,10 +145,10 @@ describeRepeats('Group Key Persistence', () => {
             })
 
             published.push(...await publishTestMessages(3))
-            const received2 = await sub2.collect(3)
+            const received2 = await sub2.collect(4)
```

Comment:
Could be worth commenting why we need to collect 4 messages instead of 3. Is the reason is that the internal group key response will be visible for the subscribers? Should it be?

Reply:
I'd add a comment if I knew why 🤷  it changed at some point. Just matching network/broker behaviour.

Reply:
But is the current behavior a bug? 

If I understood correctly, the use case we are testing is:
- subscribe with `subscriber1` 
- publish 1 test message
- expect `subscriber1` to receive the test message
- subscribe with `subscriber2`
- publish 3 more test messages
- expect `subscriber2` to receive 4 messages (`expect(received2).toEqual(published)`)

As the first message was published before `subscriber2` subscribed, should it receive the first test message (published at line 130). I'd assume it should it just receive 3 messages which were published after it subscriber.

Maybe the test would be more readable if we didn't alter the `const published` variable, but instead have variables like `published1` and `published2`.